server/src/main/java/io/crate/analyze/AlterTableDropColumnAnalyzer.java

```diff
@@ -111,6 +111,10 @@ private static void validateStatic(DocTableInfo tableInfo, List<DropColumn> drop
             }
             uniqueSet.add(colToDrop);
 
+            if (colToDrop.isSystemColumn()) {
+                throw new IllegalArgumentException("Dropping a system column is not allowed");
+            }
+
```

Comment:
Let's move this check  above. 

Otherwise  `alter table t drop column _id` fails with 
`UnsupportedFeatureException[Dropping column: _id which is part of the PRIMARY KEY is not allowed]`
which is true but _id being system column in the first place is more important.

Reply:
Thanks for reviewing! I think that is the current behaviour now, I will marked resolved.